javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,209 @@
+
+const allUpperCase = (arr) => {
+  const newUpperStringArray = [];
+  arr.forEach(string => {
+    newUpperStringArray.push(string.toUpperCase());
+  });
+  return newUpperStringArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named `greeting` that takes in a single string and returns the string in all uppercase letters, and followed by an "!".
+
+Then, write a function named `speaker` that takes in an array of strings and a callback function.
+
+Use `forEach` to build a new array of strings, each string modified by the callback. Return the new array.
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => {
+  let wordString = '';
+  wordString = word.toUpperCase() + '!';
+  return wordString;
+};
+
+const speaker = (words, callback) => {
+  const wordsStrings = [];
+  words.forEach(newWord => {
+    wordsStrings.push(callback(newWord));
+  });
+  return wordsStrings;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => {
+  arr.push(value);
+};
+
+const addNumbers = (num, arr, times, callback) => {
+  for (let i = 0; i < times; i++){
+    callback(arr, num);
+  }
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+
+CHALLENGE 6
+
+Write a function named createList that takes in an array of the current store intentory.
+
+The inventory is formatted like this:
+[
+  { name: 'apples', available: true },
+  { name: 'pears', available: true },
+  { name: 'oranges', available: false },
+  { name: 'bananas', available: true },
+  { name: 'blueberries', available: false }
+]
+
+This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
+------------------------------------------------------------------------------------------------ */
+
+const createList = (availableItems) => {
+  const finalList = [];
+  // check for truthyness
```

Comment:
Nice work here!

I really like your use of comments as it shows how you were thinking through rhw challenge to find a solution 👍